src/db_parser.jl

```diff
@@ -63,545 +353,1075 @@ function map_prime_mover(prime_mover::String)
+    attributes::Dict{Int64, Dict{String, Any}},
+    db::SQLite.DB,
+)
+    for rec in DBInterface.execute(db, QUERIES[:technologies])
+
+        # Select appropriate parametric type
+        parametric = map_prime_mover_to_parametric(rec.prime_mover)
+
+        if get_aggregation(p) == PSY.ACBus
+            area_id =
+                first(DBInterface.execute(db, QUERIES[:zone_for_technology], [rec.area])).id
+            regions = [
+                get_region(Node, p, row.name) for
+                row in DBInterface.execute(db, QUERIES[:area_to_topology], [area_id])
+            ]
+        else
+            regions = [get_region(Zone, p, rec.area)]
         end
 
-        # Calculate the average values for each day and create 24-hour profiles
-        for (date, vals) in daily_values
-            avg_value = sum(vals) / length(vals)
-            for hour in 1:24
-                push!(timestamps, "$date-$hour")
-                push!(values, avg_value)
-            end
+        if rec.prime_mover == "STORAGE"
+            t = StorageTechnology{parametric}(;
+                name=rec.prime_mover * string(rec.id),
+                id=rec.id,
+                capital_costs_discharge=LinearCurve(0.0),
+                prime_mover_type=map_prime_mover(rec.prime_mover),
+                storage_tech=map_storage_tech(rec.fuel),
+                region=regions,
+                financial_data=DEFAULT_FINANCIAL_DATA,
+                available=true,
+                power_systems_type=string(parametric),
+                operation_costs=StorageCost(),
+            )
+        elseif rec.prime_mover in ["HYDRO", "ROR", "SYNC_COND"]
+            @warn "Technologies of type $(rec.prime_mover) are not currently supported in portfolios. Skipping serialization."
+            continue
+        else
+            t = SupplyTechnology{parametric}(;
+                name=rec.prime_mover * string(rec.id),
+                id=rec.id,
+                capital_costs=LinearCurve(0.0),
+                prime_mover_type=map_prime_mover(rec.prime_mover),
+                fuel=[map_fuel(rec.fuel)],
+                co2=Dict(map_fuel(rec.fuel) => 0.0),
+                region=regions,
+                financial_data=DEFAULT_FINANCIAL_DATA,
+                available=true,
+                power_systems_type=string(parametric),
+                operation_costs=ThermalGenerationCost(
+                    variable=CostCurve(LinearCurve(0.0)),
+                    fixed=0.0,
+                    start_up=0.0,
+                    shut_down=0.0,
+                ),
+            )
         end
+        add_technology!(p, t)
     end
+end
 
-    # Parse timestamps into DateTime objects
-    parsed_timestamps = DateTime.(timestamps, "yyyy-m-d-H")
+function add_generation_units!(
+    p::Portfolio,
+    attributes::Dict{Int64, Dict{String, Any}},
+    db::SQLite.DB,
+)
+    for rec in DBInterface.execute(db, QUERIES[:generation_units])
+        component_type = eval(
```

Comment:
There are better ways to do this. We had to solve this in IS. This pattern exposes us to malicious inputs. If you get the module, you can call `getproperty` to get the type. https://github.com/NREL-Sienna/InfrastructureSystems.jl/blob/main/src/utils/utils.jl#L427

It works as long as the module has been imported, which should be true here.